massiv/src/Data/Massiv/Array/Ops/Construct.hs

```diff
@@ -120,3 +128,93 @@ enumFromStepN comp !from !step !sz = makeArray comp sz $ \ i -> from + fromInteg
 {-# INLINE enumFromStepN #-}
 
 
+-- | Function that expands an array to one with a higher dimension.
+--
+-- This is useful for constructing arrays where there is shared computation
+-- between multiple cells.  The makeArray method of constructing arrays:
+--
+-- >> makeArray :: Construct r ix e => Comp -> ix -> (ix -> e) -> Array r ix e
+--
+-- Runs a function `ix -> e` at every array index. This is inefficient if there
+-- is a substantial amount of repeated computation that could be shared while
+-- constructing elements on the same dimension, Also since there are no monadic
+-- construction functions yet that use `ix -> m e`, this can't be accomplished
+-- by using some kind of IORef approach. The expand family of functions make
+-- this possible. First you construct an `Array r (Lower ix) a` of one fewer
+-- dimensions where `a` is something like `Vector a` or `Array r Ix1 a`. Then
+-- you use 'expandWithin`' and a creation function `a -> Int -> b) to create an
```

Comment:
Shouldn't it be:
```
`expandWithin'`
```
instead of
```
'expandWithin`'
```
Also  formatting is off here (note tick and parenthesis):
```
`a -> Int -> b)
```

Reply:
Actually, now that you mention it I remembered that haddock doesn't use the back tick syntax.  I looked at the generated docs and fixed everything appropriately.

Reply:
Yeah, it does. Look at the `Index` in haddock for example: https://www.stackage.org/haddock/lts-12.22/massiv-0.2.5.0/Data-Massiv-Core-Index.html#t:Ix0
It seems like a new feature that wasn't there in older versions, not sure when they added it :)

Reply:
But it does it for links, not for code formatting, so yea, that was a right fix

Reply:
It looks like back ticks work to generate another haddock link, but they don't work to denote code.

Reply:
So \`makeArray\` does the right thing, but \`Array r Ix1 a\` does not.